sdk/src/ordinals/indexer.ts

```diff
@@ -0,0 +1,325 @@
+/**
+ * Base path for Ordinals regtest Indexer.
+ * @default "http://localhost:3003"
+ */
+export const REGTEST_ORD_BASE_PATH = "http://0.0.0.0:80";
+
+
+/**
+ * @ignore
+ */
+export interface InscriptionsData {
+    inscriptions: string[];
+    prev: string;
+    next: string;
+    lowest: number;
+    highest: number;
+}
+
+/**
+ * @ignore
+ */
+export interface InscriptionUTXO {
+    value: number;
+    script_pubkey: string,
+    address:string,
+    transaction:string,
+    sat_ranges: string,
+    inscriptions: string[];
+}
+
+/**
+ * @ignore
+ */
+export interface InscriptionSat {
+    decimal: string,
+    degree: string,
+    name: string,
+    block: number,
+    cycle: number,
+    epoch: number,
+    period: number,
+    offset: number,
+    rarity: string,
+    percentile: string,
+    satpoint: string,
+    timestamp: number,
+    inscriptions: string[];
+}
+
+/**
+ * @ignore
+ */
+export interface InscriptionDataFromId {
+    children: string[];
+    content_length: string,
+    content_type: string,
+    genesis_fee: number,
+    genesis_height: number,
+    inscription_id: string,
+    inscription_number: number,
+    next: string,
+    output_value: string,
+    parent: string,
+    previous: string,
+    rune: string,
+    sat: string,
+    satpoint: string
+    timestamp: number
+}
+
+export interface OrdinalsClient {
```

Comment:
All arguments are strings here, that doesn't seem very robust. In some cases, it doesn't make a lot of sense (like for blocknumber), and for others we can probably use stronger (more restrictive) types

Reply:
Changed